src/core/core.tooltip.js

```diff
@@ -810,9 +811,10 @@ var exports = Element.extend({
 			ctx.fillStyle = textColor;
 			helpers.each(bodyItem.before, fillLineOfText);
 
-			helpers.each(bodyItem.lines, function(line) {
+			lines = bodyItem.lines;
+			for (j = 0, len = lines.length; j < len; j++) {
 				// Draw Legend-like boxes if needed
-				if (drawColorBoxes) {
+				if (drawColorBoxes && j === 0) {
```

Comment:
actually, instead of checking if `j === 0` can we just move this `if` check to be right before the `for` loop instead of inside it?

Reply:
I made a whole mess here
the wrong `each` was converted to `for`

Reply:
Nevermind, it's ok after all will move the check.